indico/modules/events/api.py

```diff
@@ -559,24 +633,38 @@ class SessionHook(SessionContribHook):
     METHOD_NAME = 'export_session'
 
 
-class SessionFetcher(SessionContribFetcher):
+class SessionFetcher(SessionContribFetcher, SerializerBase):
     DETAIL_INTERFACES = {
         'contributions': ISessionMetadataWithContributionsFossil,
         'subcontributions': ISessionMetadataWithSubContribsFossil,
     }
 
-    def session(self, idlist):
-        ch = ConferenceHolder()
-        event = ch.getById(self._eventId)
-
-        def _iterate_objs(objIds):
-            for objId in objIds:
-                session = event.getSessionById(objId)
-                for obj in session.getSlotList():
-                    if obj is not None:
-                        yield obj
+    def __init__(self, aw, hook):
+        super(SessionFetcher, self).__init__(aw, hook)
+        self.user = getattr(aw.getUser(), 'user', None)
 
-        return self._process(_iterate_objs(idlist))
+    def session(self, idlist):
+        event = Event.get_one(self._eventId)
```

Comment:
Did we really fail with an exception before if the event id did not exist or did we simply return an empty response?